tests/lib.rs

```diff
@@ -0,0 +1,700 @@
+}
+
+#[ignore = "Relies on a running memcached server"]
+#[tokio::test]
+async fn test_delete_no_reply() {
+    let key = "async-memcache-test-key-delete-no-reply";
+
+    let mut client = setup_client(vec![key]).await;
+
+    let value = format!("{}", rand::random::<u64>());
+    let set_result = client.set(key, value.as_str(), None, None).await;
+
+    assert!(
+        set_result.is_ok(),
+        "failed to set {}, {:?}",
+        key,
+        set_result
+    );
+
+    let get_result = client.get(key).await;
+
+    assert!(
+        get_result.is_ok(),
+        "failed to get {}, {:?}",
+        key,
+        get_result
+    );
+
+    let get_result = get_result.unwrap();
+
+    match get_result {
+        Some(get_value) => assert_eq!(
+            String::from_utf8(get_value.data).expect("failed to parse a string"),
+            value
+        ),
+        None => panic!("failed to get {}", key),
+    }
+
+    let delete_result = client.delete_no_reply(key).await;
+
+    assert!(
+        delete_result.is_ok(),
+        "failed to delete {}, {:?}",
+        key,
+        delete_result
+    );
+}
+
+#[ignore = "Relies on a running memcached server"]
+#[tokio::test]
+async fn test_set_multi_with_string_values() {
+    let keys = vec!["smwsv-key1", "smwsv-key2", "smwsv-key3"];
+    let values = vec!["value1", "value2", "value3"];
+
+    let kv: Vec<(&str, &str)> = keys.clone().into_iter().zip(values.into_iter()).collect();
+
+    let mut client = setup_client(keys).await;
+
+    let _ = client.set_multi(kv, None, None).await;
+
+    let result = client.get("smwsv-key2").await;
+
+    assert!(matches!(
+        std::str::from_utf8(
+            &result
+                .expect("should have unwrapped a Result")
+                .expect("should have unwrapped an Option")
+                .data
+        )
+        .expect("failed to parse string from bytes"),
+        "value2"
+    ));
+}
+
+#[ignore = "Relies on a running memcached server"]
+#[tokio::test]
+async fn test_set_multi_with_string_values_that_exceed_max_size() {
+    const NUM_PAIRS: usize = 100;
+    const LARGE_VALUE_SIZE: usize = 2_048_576;
+    const NUM_LARGE_KEYS: usize = 5;
+
+    let keys: Vec<String> = (0..NUM_PAIRS).map(|i| format!("multi-key{}", i)).collect();
+    let mut values: Vec<String> = (0..NUM_PAIRS).map(|i| format!("value{}", i)).collect();
+
+    let mut rng = rand::thread_rng();
+    let large_key_indices: Vec<usize> = (1..NUM_PAIRS)
+        .choose_multiple(&mut rng, NUM_LARGE_KEYS)
+        .into_iter()
+        .collect();
+
+    let mut large_key_strs = Vec::new();
+    for &index in &large_key_indices {
+        values[index] = "a".repeat(LARGE_VALUE_SIZE);
+        large_key_strs.push(keys[index].clone());
+    }
+
+    let kv: Vec<(&str, &str)> = keys
+        .iter()
+        .map(AsRef::as_ref)
+        .zip(values.iter().map(AsRef::as_ref))
+        .collect();
+
+    let mut client = setup_client(keys.iter().map(AsRef::as_ref).collect()).await;
```

Comment:
This test is used to show that responses in the result HashMap are properly mapped to their corresponding keys